pkg/kvstore/etcd.go

```diff
@@ -81,6 +81,9 @@ func newEtcdClient(config *client.Config, cfgPath string) (KVClient, error) {
 		}
 	}
 	if config != nil {
+		if config.DialTimeout == 0 {
+			config.DialTimeout = 10 * time.Second
```

Comment:
1 minute is fine. 

Let's say a user forgets to start etcd.  Unless the user checks that etcd is
around they might not notice the agent is waiting for them to do something.
Failing hard after a timeout as the default behaviour is friendlier IMO.

Please let me know if I misunderstood https://github.com/cilium/cilium/issues/1234

Thanks.